source/main.c

```diff
@@ -0,0 +1,242 @@
+#include <switch.h>
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <arpa/inet.h>
+#include <sys/socket.h>
+#include <sys/time.h>
+#include <errno.h>
+#include <netdb.h>
+#include <unistd.h>
+
+#define PORT 8192
+#define CPU_CLOCK 250000000 //250 MHz should be a BIG improvement in battery life if the console is used in handheld mode
+
+char ipAddress[16];
+u8 data[5];
+JoystickPosition joystickLeft, joystickRight;
+
+void * get_in_addr(struct sockaddr *sa) {
+  if (sa->sa_family == AF_INET) {
+    return &(((struct sockaddr_in*)sa)->sin_addr);
+  }
+
+  return &(((struct sockaddr_in6*)sa)->sin6_addr);
+}
+
+int broadcast(int sck, char * host, char * port) {
+  uint8_t msg[] = "xbox_switch";
+  char buffer[128] = {0};
+  struct hostent * he;
+  struct sockaddr_in remote;
+  struct sockaddr_storage from;
+  socklen_t from_len = sizeof(from);
+
+  if ((he = gethostbyname(host)) == NULL) {
+    return -1;
+  }
+
+  remote.sin_family = AF_INET;
+  remote.sin_port = htons(atoi(port));
+  remote.sin_addr = *(struct in_addr *)he->h_addr;
+  memset(remote.sin_zero, 0, sizeof(remote.sin_zero));
+
+  int on=1;
+  setsockopt(sck, SOL_SOCKET, SO_BROADCAST, &on, sizeof(on));
+
+  struct timeval tv;
+  tv.tv_sec = 0;
+  tv.tv_usec = 1000000;
+  setsockopt(sck, SOL_SOCKET, SO_RCVTIMEO,&tv,sizeof(tv));
+
+  while(1) {
+    while (1) {
+      sendto(sck, msg, sizeof(msg), 0, (struct sockaddr *)&remote, sizeof(remote));
+      recvfrom(sck, buffer, sizeof(buffer), 0, (struct sockaddr *)&from, &from_len);
+
+      if (strlen(buffer) > 0) break;
+    }
+
+    char addr[40] = {0};
+
+    const char * ptr = inet_ntop(from.ss_family, get_in_addr((struct sockaddr *)&from), addr, sizeof(addr));
+
+    if (strcmp("xbox", buffer) == 0) {
+      strcpy(ipAddress, ptr);
+      break;
+    }
+  }
+
+  return 0;
+}
+
+int main(int argc, char* argv[]) {
+	u8 ipBlock1 = 192;
+	u8 ipBlock2 = 168;
+    u8 ipBlock3 = 1;
+    u8 ipBlock4 = 255;
+    u8 currentIpBlock = 0;    
+
+    socketInitializeDefault();
+    consoleInit(NULL);
+
+    printf("|------------------------|\n");
+    printf("| Switch XBOX Controller |\n");
+    printf("|------------------------|\n\n");
+
+    printf("Please set up the ip you would like to send the UDP broadcast too\n.");
+    printf("Usally this address is 192.168.X.255 where X is the 3rd block of your local IP.\n");
+    printf("If UDP broadcasting does not work for you, use the IP of your computer instead.\n");
+
+	consoleUpdate(NULL);
+
+	pcvSetClockRate(PcvModule_Cpu, CPU_CLOCK); //Underclock the CPU
+	appletSetScreenShotPermission(0); //Disable the screenshot function because it is not needed for the program
+	appletBeginBlockingHomeButton(0); //Block the Home Button to prevent acidental use. Mainly useful for the pro controller since (+) and Home are close together.
```

Comment:
Have to remove that one. Will lock the console if it is ran from the hbmenu nsp.